src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -594,6 +602,117 @@ public void respawn()
+	{
+		Preconditions.checkNotNull(location, "Location cannot be null");
+		Preconditions.checkArgument(hand == EquipmentSlot.HAND || hand == EquipmentSlot.OFF_HAND, "Must be HAND or OFF_HAND");
+
+		ItemStack itemInHand = hand == EquipmentSlot.HAND ? getItemInHand() : getInventory().getItemInOffHand();
+
+		boolean cancelled = false;
+		BlockMock block = BlockMock.mock(location.getBlock());
+
+		if (getGameMode() == GameMode.SPECTATOR)
+		{
+			cancelled = !(block.getState() instanceof Container);
+		}
+
+		// TODO Implement Cooldown
+//		if (hasCooldown(itemInHand.getType())) {
+//			cancelled = true;
+//		}
+
+		ItemStack eventItemStack = itemInHand;
+		if (eventItemStack.getType() == Material.AIR || eventItemStack.getAmount() == 0)
+		{
+			eventItemStack = null;
+		}
+
+		PlayerInteractEvent event = new PlayerInteractEvent(this, Action.RIGHT_CLICK_BLOCK, eventItemStack, block, clickedFace, hand);
+		if (cancelled)
+		{
+			event.setUseInteractedBlock(Event.Result.DENY);
+		}
+		Bukkit.getServer().getPluginManager().callEvent(event);
+
+		boolean interactResult = event.useItemInHand() == Event.Result.DENY;
+
+		if (event.useInteractedBlock() == Event.Result.DENY)
+		{
+			return event;
+		}
+		else if (getGameMode() == GameMode.SPECTATOR)
+		{
+			if (block.getState() instanceof ContainerMock containerMock)
+			{
+				openInventory(containerMock.getInventory());
+			}
+		}
+		else
+		{
+			boolean isItemInHand = !getItemInHand().isEmpty() || !getInventory().getItemInOffHand().isEmpty();
+			boolean shouldCancelUse = isSneaking() && isItemInHand;
+			InteractionResult interactionResult;
+
+			if (!shouldCancelUse)
+			{
+				if (!(block.getState() instanceof BlockStateMock blockStateMock))
+				{
+					throw new UnimplementedOperationException();
+				}
+
+				ItemInteractionResult itemInteractionResult = blockStateMock.simulateUseItemOn(blockStateMock, location, ItemStackMock.mock(getItemInHand()), this, hand, clickedFace);
+				block.setState(blockStateMock);
+				block.setBlockData(blockStateMock.getBlockData());
+
+				if (itemInteractionResult.consumesAction())
+				{
+					// TODO implement CriterionTriggers
+					return event;
+				}
+
+				if (itemInteractionResult == ItemInteractionResult.PASS_TO_DEFAULT_BLOCK_INTERACTION && hand == EquipmentSlot.HAND) {
+					interactionResult = blockStateMock.simulateUseWithoutItem(blockStateMock, location, this, clickedFace);
+					block.setState(blockStateMock);
+					block.setBlockData(blockStateMock.getBlockData());
+					if (interactionResult.consumesAction()) {
+						return event;
+					}
+				}
+			}
+
+			if (!itemInHand.isEmpty() && !interactResult)
+			{
+				ItemStackMock itemStackMock = ItemStackMock.mock(itemInHand);
+				int count = itemStackMock.getAmount();
+
+				InteractionResult result = itemStackMock.simulateUseItemOn(this, location, hand);
```

Comment:
This will always throw an UnimplementedOperationException. Everything inside this if code segment is reduntant